### ISSUE TITLE
command-not-found handling on Fedora

### ISSUE THREAD

**Issue by glennj** [fish-shell/fish-shell#1918]
$ fish --version
    fish, version 2.1.1-953-ga0e5cdd

For any command-not-found, I see this:

    $ foobar
    bash: --: command not found...

First, it says "bash". Boo.
Second, where is this `--` coming from?

In `__fish_config_interactive`, we have:

```fish
                # Check for Fedora's handler
                else if test -f /usr/libexec/pk-command-not-found
                        function __fish_command_not_found_handler --on-event fish_command_not_found
                                /usr/libexec/pk-command-not-found -- $argv
                        end
```

Please, remove the incorrect `--` argument:

    $ /usr/libexec/pk-command-not-found -- foobar
    bash: --: command not found...
    $ /usr/libexec/pk-command-not-found foobar
    bash: foobar: command not found...

Second, "bash" seems to be hard-coded in pk-command-not-found, so remove it

```fish
function __fish_command_not_found_handler --on-event fish_command_not_found
	/usr/libexec/pk-command-not-found $argv[1] ^&1  | sed 's/^bash: //' >&2
	return 127
end
```

**Comment by zanchey** [comment]
Good news! The first problem is already fixed in Git master - see #1900.

I'm not so sure about the second change. Perhaps that should be taken up with the [PackageKit](https://github.com/hughsie/PackageKit) developers? I see [a request to improve the output for zsh in Fedora Bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=678934) was orphaned, but we might have more luck with Github.

**Comment by zanchey** [comment]
This got fixed in PackageKit upstream as https://github.com/hughsie/PackageKit/pull/60, although it's not in any released versions yet. The change is not perfect - it assumes the running shell is the login shell - but I'm going to close this for now as I don't think there's anything more we can do here.
